### PR TITLE
Popups & clickstream fixes

### PR DESCRIPTION
**Pull request by rich-nguyen**
Clickstream refactoring. 

Also there were two issues to fix that caused #9343 . First, a clearfix was styling the dropdown behind the live blog content. Second, there were clickstream changes which caused the toggles component to think that the order by click was a non-component click, hence the dropdown was immediately shown then reset. 

Chrome dev tools' purple highlighting showed the DOM changed quicker than the naked eye could see, nice.

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/ui/clickstream.js [hunk 1]
```diff
@@ -45,34 +45,30 @@ define([
...
+
+                    if (spec.validTarget && el.getAttribute('data-link-test')) {
+                        spec.tag = el.getAttribute('data-link-test') + ' | ' + spec.tag;
                     }
+                    return spec;
                 }
 
                 if (!spec.validTarget) {
-                    spec.validTarget = filterSource(el.tagName.toLowerCase()).length > 0 || forceValid;
+                    spec.validTarget = filterSource(el.tagName.toLowerCase()).length > 0 || !!forceValid;
```
Comment (piuccio) [inline static/src/javascripts/projects/common/modules/ui/clickstream.js]: you can reuse `elName` defined above

Comment (rich-nguyen) [inline static/src/javascripts/projects/common/modules/ui/clickstream.js]: yep, good spot

**Comment by piuccio** [comment]
Nice work. The spec doesn't seem to test the behaviour of `body` which has changed, but I trust you it's fine :smile: 

:+1:

File: static/src/javascripts/projects/common/utils/$.js [hunk 2]
```diff
@@ -24,11 +26,25 @@ define([
             return $.ancestor(el.parentNode, c);
         }
     };
+
+    $.isDescendantOrSelf = function (el, ancestor) {
+        if (!el || !ancestor || !el.parentNode || el.nodeName.toLowerCase() === 'html') {
+            return false;
+        }
+
+        if (el === ancestor || el.parentNode === ancestor) {
```
Comment (robertberry) [inline static/src/javascripts/projects/common/utils/$.js]: Why do you check `parentNode` here? Won't it be checked in the next call of the function?

**Comment by robertberry** [comment]
:+1:

File: static/src/javascripts/projects/common/utils/$.js (hunk 2 above)
Comment (piuccio) [inline static/src/javascripts/projects/common/utils/$.js]: and how is `$.isDescendantOrSelf` different from `bonzo.isAncestor`?

**Pull request merged**